### PR TITLE
[improve][io] support array schema for JDBC postgres connector

### PR DESCRIPTION
**Pull request by freeznet**
Fixes #xyz

Main Issue: #xyz

PIP: #xyz 

### Motivation

Currently, the JDBC PostgreSQL connector in Apache Pulsar doesn't support array schema types, limiting its ability to handle complex data structures. Many real-world applications require storing array data in PostgreSQL databases (e.g., tags, categories, multi-valued attributes). 

Without this support, users have to:
- Manually serialize arrays into strings or JSON
- Use workarounds that don't leverage PostgreSQL's native array capabilities
- Lose type safety and query performance benefits of native array columns

This PR addresses these limitations by adding comprehensive array schema support for the PostgreSQL JDBC connector.

### Modifications

1. **Core PostgreSQL Array Implementation** (`PostgresJdbcAutoSchemaSink.java`):
   - Added array type conversion logic for all common PostgreSQL array types (INTEGER[], TEXT[], BOOLEAN[], NUMERIC[], REAL[], BIGINT[], etc.)
   - Implemented proper validation and error handling for array elements
   - Added type mapping between Avro array schemas and PostgreSQL array types
   - Ensures type consistency across all array elements

2. **Framework Enhancement** (`BaseJdbcAutoSchemaSink.java`):
   - Extended the base JDBC sink to support array handling
   - Added abstract method `handleArrayValue` for database-specific implementations
   - Maintained strict backward compatibility - existing functionality remains unchanged

3. **Database-Specific Support** (`SqliteJdbcAutoSchemaSink.java`):
   - Added explicit error messaging for SQLite to indicate array non-support
   - Provides clear guidance to users when attempting to use arrays with unsupported databases

4. **Testing Infrastructure**:
   - Created comprehensive test utilities (`PostgresArrayTestUtils.java`) for generating test array data
   - Added full integration test suite (`PostgresJdbcArrayIntegrationTest.java`) covering:
     - All supported array types
     - Empty arrays
     - Arrays with null elements
     - Boundary value testing
     - End-to-end data flow validation

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- Documentation should be updated to include:
  - Supported array types for PostgreSQL JDBC connector
  - Configuration examples for array schemas
  - Limitations and best practices

- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/freeznet/pulsar/pull/13

### CODE REVIEW

File: pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/BaseJdbcAutoSchemaSink.java [hunk 1]
```diff
@@ -174,10 +231,73 @@ private static void setColumnNull(PreparedStatement statement, int index, int ty
...
         log.debug("Setting column value, statement: {}, index: {}, value: {}", statement, index, value);
 
+        // Handle null values first
+        if (value == null) {
+            setColumnNull(statement, index, java.sql.Types.NULL);
+            return;
+        }
+
+        // Check for array types first, before other type checks
+        if (value instanceof GenericData.Array) {
```
Comment (omri-cavnue) [inline pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/BaseJdbcAutoSchemaSink.java]:
```suggestion
        if (value instanceof GenericData.Array || value instanceof Object[]) {
```

Comment (omri-cavnue) [reply pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/BaseJdbcAutoSchemaSink.java]: Minor change needed to enable numeric types

**Review by omri-cavnue — APPROVED** [review]
